src/util/heap/accounting.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+
+/// The struct is used for page usage.
+/// Both page resource and side metadata uses this struct to do page accounting.
+pub struct PageAccounting {
+    /// The reserved pages. This should be incremented when we are about to allocate pages.
+    /// Note this is different than quarantining address range. We do not count for quarantined
+    /// memory.
+    reserved: AtomicUsize,
+    /// The committed pages. This should be incremented when we succesfully allocate pages from the OS.
+    committed: AtomicUsize,
+}
+
+impl PageAccounting {
+    pub fn new() -> Self {
+        Self {
+            reserved: AtomicUsize::new(0),
+            committed: AtomicUsize::new(0),
+        }
+    }
+
+    pub fn reserve_and_commit(&self, pages: usize) {
```

Comment:
Probably a short comment for each of these methods will be useful

Reply:
Done.